note-check: need skip deleted file

backgrounds:

```bash
2024/07/08 07:48:27.307141 [3b99ce30-3cfe-11ef-9397-5607aa39a00c][INFO] server.go:138: [note-check] config on repo qbox/net-sophon: Linter{Enable: true, WorkDir: /var/tmp/gitrepo2514953585, Command: [note-check], Args: [], ReportFormat: github_check_run, ConfigPath: }
2024/07/08 07:48:27.336201 [3b99ce30-3cfe-11ef-9397-5607aa39a00c][ERROR] server.go:155: failed to run linter: open /var/tmp/gitrepo2514953585/common/clients/dynnode/client.go: no such file or directory
2024/07/08 07:48:27.336230 [3b99ce30-3cfe-11ef-9397-5607aa39a00c][INFO] server.go:138: [commit-check] config on repo qbox/net-sophon: Linter{Enable: true, WorkDir: /var/tmp/gitrepo2514953585, Command: [commit-check], Args: [], ReportFormat: github_check_run, ConfigPath: }
```

Related file:
<img width="936" alt="image" src="https://github.com/qiniu/reviewbot/assets/10810759/4506d439-e6de-406c-b3fe-52bc7de24653">

--- a/internal/linters/doc/note-check/note.go
+++ b/internal/linters/doc/note-check/note.go
@@ -31,7 +31,7 @@ func noteCheckHandler(log *xlog.Logger, a linters.Agent) error {
 	for _, file := range a.PullRequestChangedFiles {
 		fileName := file.GetFilename()
 		// Only check go files
-		if filepath.Ext(fileName) != ".go" {
+		if filepath.Ext(fileName) != ".go" || file.GetStatus() == "removed" {
 			continue
 		}
 

